sv_pipeline/genome/load_data_tests.py

```diff
@@ -0,0 +1,279 @@
+TEST_INPUT_DATASET = 'test_dataset/input_vcf.gz'
+TEST_MT_PATH = 'test_mt/data.mt'
+TEST_GENCODE_PATH = 'test_gtf/gtf'
+TEST_HOST = 'TEST_HOST'
+TEST_PORT = '9500'
+TEST_GENCODE_RELEASE = 31
+TEST_BLOCK_SIZE = 3000
+TEST_NUM_SHARDS = 10
+TEST_INDEX_NAME = 'test_guid__structural_variants__wgs__grch38__20210426'
+
+
+class LoadDataTest(unittest.TestCase):
+    def setUp(self):
+        self.vcf_file = tempfile.mkstemp(suffix='.vcf')[1]
+        with open(self.vcf_file, 'w') as f:
+            f.writelines('\n'.join(VCF_DATA))
+        hl.init()
+        self.mt = hl.import_vcf(self.vcf_file, reference_genome='GRCh38')
+
+    def tearDown(self):
+        hl.stop()
+        os.remove(self.vcf_file)
+
+    @mock.patch('sv_pipeline.genome.load_data.os')
+    @mock.patch('sv_pipeline.genome.load_data.hl')
+    @mock.patch('sv_pipeline.genome.load_data.logger')
+    def test_load_mt(self, mock_logger, mock_hl, mock_os):
+        mock_os.path.splitext.return_value = ['test_dataset/input_vcf', 'gz']
+        mock_os.path.isdir.return_value = True
+        _ = load_mt(TEST_INPUT_DATASET, '', False)
+        mock_hl.read_matrix_table.assert_called_with('test_dataset/input_vcf.mt')
+        mock_logger.info.assert_called_with('Use the existing MatrixTable file test_dataset/input_vcf.mt. '
+                                       'If the input VCF file has been changed, or you just want to re-import VCF,'
+                                       ' please add "--overwrite-matrixtable" command line option.')
+
+
+    @mock.patch('sv_pipeline.genome.load_data.get_sample_subset')
+    @mock.patch('sv_pipeline.genome.load_data.get_sample_remap')
+    @mock.patch('sv_pipeline.genome.load_data.logger')
+    def test_subset_mt(self, mock_logger, mock_get_remap, mock_get_sample):
+        mock_get_sample.return_value = {'SAMPLE-1', 'SAMPLE-3', 'SAMPLE-5', 'SAMPLE-6'}
+        with self.assertRaises(Exception) as e:
+            _ = subset_mt('test_guid', self.mt, skip_sample_subset=False, ignore_missing_samples=False)
+        self.assertEqual(str(e.exception), 'Missing the following 1 samples:\nSAMPLE-6')
+
+        mock_get_remap.return_value = {'SAMPLE-1': 'SAMPLE-1-REMAP'}
+        rows = subset_mt('test_guid', self.mt, skip_sample_subset=False, ignore_missing_samples=True)
+        calls = [
+            mock.call('Missing the following 1 samples:\nSAMPLE-6'),
+            mock.call('Subsetting to 4 samples (remapping 1 samples)'),
+        ]
+        mock_logger.info.assert_has_calls(calls)
+        mock_get_sample.assert_called_with('test_guid', 'WGS')
+        mock_get_remap.assert_called_with('test_guid', 'WGS')
+        self.assertEqual(rows.count(), 5)
+        self.assertEqual(set(rows.genotypes.map(lambda x: x.sample_id).take(1)[0]), {'SAMPLE-1-REMAP', 'SAMPLE-3', 'SAMPLE-5'})
+
+        mock_logger.reset_mock()
+        rows = subset_mt('test_guid', self.mt, skip_sample_subset=True)
+        self.assertEqual(rows.count(), 5)
```

Comment:
you should add a test that rows get filtered out properly if none of the subsetted samples are present

Reply:
If none of the samples found, there will be no variant rows in the output. Is that behavior we want?